oidc-service-impl/src/main/java/io/mosip/esignet/services/LinkedAuthorizationServiceImpl.java

```diff
@@ -224,17 +210,19 @@ public LinkedKycAuthResponse authenticateUser(LinkedKycAuthRequest linkedKycAuth
     @Override
     public LinkedConsentResponse saveConsent(LinkedConsentRequest linkedConsentRequest) throws EsignetException {
         OIDCTransaction transaction = cacheUtilService.getLinkedAuthTransaction(linkedConsentRequest.getLinkedTransactionId());
-        if(transaction == null) {
+        if(transaction == null || ConsentAction.NOCAPTURE.equals(transaction.getConsentAction())) {
             throw new InvalidTransactionException();
         }
-
-        authorizationHelperService.validateAcceptedClaims(transaction, linkedConsentRequest.getAcceptedClaims());
-        authorizationHelperService.validateAuthorizeScopes(transaction, linkedConsentRequest.getPermittedAuthorizeScopes());
+        List<String> acceptedClaims = linkedConsentRequest.getAcceptedClaims();
+        List<String> permittedAuthorizeScopes = linkedConsentRequest.getPermittedAuthorizeScopes();
+        authorizationHelperService.validateAcceptedClaims(transaction, acceptedClaims);
+        authorizationHelperService.validateAuthorizeScopes(transaction, permittedAuthorizeScopes);
         // cache consent only, auth-code will be generated on link-auth-code-status API call
         transaction.setAcceptedClaims(linkedConsentRequest.getAcceptedClaims());
         transaction.setPermittedScopes(linkedConsentRequest.getPermittedAuthorizeScopes());
-        cacheUtilService.setLinkedConsentedTransaction(linkedConsentRequest.getLinkedTransactionId(), transaction);
 
+        cacheUtilService.setLinkedConsentedTransaction(linkedConsentRequest.getLinkedTransactionId(), transaction);
+        cacheUtilService.setLinkedConsentTransaction(linkedConsentRequest.getLinkedTransactionId(), transaction);
```

Comment:
why do we store in both ?

Reply:
I'll rename the second one.
the first one is referred by another service to trigger kafka message
second is used to store consent in DB